packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -287,13 +294,449 @@ class HeaderMatchImpl extends HttpHeaderMatch {
+   */
+  static valueDoesNotStartWith(metadataName: string, prefix: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, true, { prefix });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must end with
+   * the specified characters.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param suffix The suffix to test against
+   */
+  static valueEndsWith(metadataName: string, suffix: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, false, { suffix });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must not end
+   * with the specified characters.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param suffix The suffix to test against
+   */
+  static valueDoesNotEndWith(metadataName: string, suffix: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, true, { suffix });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must include
+   * the specified characters.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param regex The regex to test against
+   */
+  static valueMatchesRegex(metadataName: string, regex: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, false, { regex });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must not
+   * include the specified characters.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param regex The regex to test against
+   */
+  static valueDoesNotMatchRegex(metadataName: string, regex: string): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, true, { regex });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must be in a
+   * range of values.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param start Match on values starting at and including this value
+   * @param end Match on values up to but not including this value
+   */
+  static valuesIsInRange(metadataName: string, start: number, end: number): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, false, {
+      range: {
+        start,
+        end,
+      },
+    });
+  }
+
+  /**
+   * The value of the metadata with the given name in the request must not be in
+   * a range of values.
+   *
+   * @param metadataName the name of the gRPC metadata to match against
+   * @param start Match on values starting at and including this value
+   * @param end Match on values up to but not including this value
+   */
+  static valuesIsNotInRange(metadataName: string, start: number, end: number): GrpcMetadataMatch {
+    return new MetadataMatchImpl(metadataName, true, {
+      range: {
+        start,
+        end,
+      },
+    });
+  }
```

Comment:
This is a copy-paste of `HttpHeaderMatch`. Can we unify these, so we only have a single class for this?

Reply:
Good point. I believe those two shares same properties intentionally and thus should be able to create single class.

Reply:
I remained `MetadataMatch` since I believe HTTP Header is technically considered as a metadata.